Adding documentation about explain in hybrid query and hybrid_score_explanation processor (#9053)
opensearch-project/documentation-website: _search-plugins/search-pipelines/explanation-processor.md

---
layout: default
title: Hybrid score explanation
nav_order: 15
has_children: false
parent: Search processors
grand_parent: Search pipelines
---

# Hybrid score explanation processor
Introduced 2.19
{: .label .label-purple }

> The `hybrid_score_explanation` response processor adds the normalization and combination results to the returned search response. For more information, see [`hybrid` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/hybrid/) that is the debugging tool that provides insights of score normalization processing.  

Reviewer: I'm not following the second sentence here. Please rephrase or delete the text following the link.

> The `hybrid_score_explanation` response processor adds the normalization and combination results to the returned search response. For more information, see [`hybrid` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/hybrid/) that is the debugging tool that provides insights of score normalization processing.  

Reviewer 2: ```suggestion
The `hybrid_score_explanation` response processor adds the normalization and combination results to the returned search response. You can use it as a debugging tool to gain insights of score normalization processing.  For more information, see [Hybrid query]({{site.url}}{{site.baseurl}}/query-dsl/compound/hybrid/).
```

## Example

The following example demonstrates using a search pipeline with a `hybrid_score_explanation` processor.

For a comprehensive example, follow the [Neural search tutorial]({{site.url}}{{site.baseurl}}/ml-commons-plugin/semantic-search#tutorial).

### Creating a search pipeline 

The following request creates a search pipeline containing a `normalization-processor` and a `hybrid_score_explanation` processor:

```json
PUT /_search/pipeline/nlp-search-pipeline
{
  "description": "Post processor for hybrid search",
  "phase_results_processors": [
    {
      "normalization-processor": {
        "normalization": {
          "technique": "min_max"
        },
        "combination": {
          "technique": "arithmetic_mean"
        }
      }
    }
  ],
  "response_processors": [
    {
        "hybrid_score_explanation": {}
    }
  ]
}
```
{% include copy-curl.html %}

### Using a search pipeline

> To see explanation details, specify `explain=true` in your search request:

Reviewer: ```suggestion
To see explanation information, specify `explain=true` in your search request:
```
